packages/react-core/src/components/DualListSelector/DualListSelector.tsx

```diff
@@ -0,0 +1,319 @@
+    });
+
+    const onOptionSelect = this.props.onOptionSelect;
+    if (onOptionSelect) {
+      onOptionSelect(e);
+    }
+  };
+
+  handleKeys = (event: KeyboardEvent) => {
+    const key = event.key;
+    let moveFocus = false;
+    let currentIndex = -1;
+    const controls = Array.from(this.controlsEl.current.getElementsByClassName('pf-c-button'));
+    if (['ArrowUp', 'ArrowDown', 'ArrowLeft', 'ArrowRight'].includes(event.key)) {
+      if (document.activeElement === this.controlsEl.current) {
+        currentIndex = 0;
+        moveFocus = true;
+      } else {
+        controls.forEach((control, index) => {
+          if (document.activeElement === control) {
+            currentIndex = key === 'ArrowUp' || key === 'ArrowLeft' ? index - 1 : index + 1;
+            moveFocus = true;
+          }
+        });
+      }
+      event.preventDefault();
+    }
+    if (moveFocus && controls[currentIndex]) {
+      (controls[currentIndex] as HTMLElement).focus();
+    }
+  };
+
+  componentDidMount() {
+    window.addEventListener('keydown', this.handleKeys);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('keydown', this.handleKeys);
+  }
+
+  render() {
+    const {
+      availableOptionsTitle,
+      availableOptionsActions,
+      className,
+      chosenOptionsTitle,
+      chosenOptionsActions,
+      filterOption,
+      isSearchable,
+      chosenOptionsStatus,
+      availableOptionsStatus,
+      /* eslint-disable @typescript-eslint/no-unused-vars */
+      availableOptions: consumerPassedAvailableOptions,
+      chosenOptions: consumerPassedChosenOptions,
+      removeSelected,
+      addAll,
+      removeAll,
+      addSelected,
+      id,
+      ...props
+    } = this.props;
+    const { availableOptions, chosenOptions, chosenOptionsSelected, availableOptionsSelected } = this.state;
+    const availableOptionsStatusToDisplay =
+      availableOptionsStatus || `${availableOptionsSelected.length} of ${availableOptions.length} items selected`;
+    const chosenOptionsStatusToDisplay =
+      chosenOptionsStatus || `${chosenOptionsSelected.length} of ${chosenOptions.length} items selected`;
+
+    return (
+      <div className={css(styles.dualListSelector, className)} id={id} {...props}>
+        <DualListSelectorPane
+          isSearchable={isSearchable}
+          filterOption={filterOption}
+          status={availableOptionsStatusToDisplay}
+          title={availableOptionsTitle}
+          options={availableOptions}
+          selectedOptions={availableOptionsSelected}
+          onOptionSelect={this.onOptionSelect}
+          actions={availableOptionsActions}
+          id={`${id}-available-pane`}
+        />
+        <div
+          className={css(styles.dualListSelectorControls)}
+          tabIndex={0}
+          ref={this.controlsEl}
+          role="listbox"
+          aria-label="Selector controls"
+        >
+          <div className={css('pf-c-dual-list-selector__controls-item')} role="option">
```

Comment:
Can we use the `styles` object for these (also on lines 270, 281, 292)? Or are they placeholder classes?

Reply:
yep, placeholder classes.